loader/src/api-client.js

```diff
@@ -23,13 +23,29 @@ class ApiClient {
   }
 
   async getLocations(query) {
-    const { body } = await httpClient({
-      url: `${this.url}/locations`,
+    let options = {
+      url: `${this.url}/api/edge/locations`,
       searchParams: query,
       headers: { "x-api-key": this.key },
       responseType: "json",
-    });
-    return body;
+    };
+
+    const results = [];
+    while (options) {
+      const { body } = await httpClient(options);
+      results.push(...body.data);
+
+      if (body.links?.next) {
+        options = {
+          ...options,
+          url: new URL(body.links.next, this.url).href,
+          searchParams: undefined,
+        };
+      } else {
+        options = null;
```

Comment:
Nah, not a big reason. This is one of those ”mentors early on pounded into me to avoid `while(true)` whenever possible so it’s just what I tend to do” things. (Unsurprisingly, during a time when I wrote more C code. 😜)

Will change.